### PR TITLE
docs: Add test & debugging doc

### PR DESCRIPTION
**Pull request by jrconlin**
Add doc describing using contract testing for local debugging.
[CONSVC-1974](https://mozilla-hub.atlassian.net/browse/CONSVC-1974)

### CODE REVIEW

**Comment by Trinaa** [comment]
@jrconlin 

As discussed, I attempted merged TESTING.MD with the contract test READMEs. Let me know your thoughts on the 'Local Execution and Debugging' sections.

I'm still pondering about adding instructions for running contract tests individually and maybe to add a 'Local Execution and Debugging' section to the client README

File: test-engineering/contract-tests/client/README.md [hunk 1]
```diff
@@ -112,4 +113,73 @@ Example:
...
+
+1. Modify `test-engineering/contract-tests/docker-compose.yml`
+
+    In the partner definition, expose port 5000 by adding the following:
+    ```yaml
+    ports:
+      - "5000:5000"
+    ```
+
+    In the Contile definition, expose port 5000 by adding the following:
```
Comment (pjenvey) [inline test-engineering/contract-tests/client/README.md:150]:
```suggestion
    In the Contile definition, expose port 8000 by adding the following:
```

File: test-engineering/contract-tests/client/README.md [hunk 2]
```diff
@@ -112,4 +113,73 @@ Example:
...
+
+   Execute the following from the project root:
+   ```text
+   docker compose -f test-engineering/contract-tests/docker-compose.yml up contile
+   ```
+
+3. Run the contract tests
+
+    Execute the following from the project root:
+    ```text
```
Comment (pjenvey) [inline test-engineering/contract-tests/client/README.md:166]: Note you could also use **```shell** on these

**Review by pjenvey — CHANGES_REQUESTED** [review]

File: test-engineering/contract-tests/README.md [hunk 3]
```diff
@@ -39,26 +40,73 @@ MTS.
...
+
+**Fetching Tiles**
+
+Contile will attempt to look up your IP address if you access it using Firefox. 
+The easiest way to get a test response back would be to craft a curl request. 
+For example (presuming that Contile is running on `http://localhost:8000/v1/tiles`):
+
+```sh
+curl -v \
+    -H "UserAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:10.0) Gecko/20100101 Firefox/91.0'" \
```
Comment (pjenvey) [inline test-engineering/contract-tests/README.md:101]: Or there's also `-A` (which I rarely use)

```suggestion
    -H "User-Agent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:10.0) Gecko/20100101 Firefox/91.0'" \
```

**Review by pjenvey — CHANGES_REQUESTED** [review]
Sorry I missed this one last week, looks good otherwise

**Review by pjenvey — APPROVED** [review]

File: test-engineering/contract-tests/README.md [hunk 4]
```diff
@@ -39,26 +40,73 @@ MTS.
...
+MTS Docker container such as a partner settings file
+- the `volumes/partner` directory contains a YML file which defines every
+response that the API returns keyed by form-factor and then os-family
+
+
+## Local Execution
+
+To run the contract tests locally, execute the following from the repository root:
+
+**Build Contile Docker Image**
```
Comment (tiftran) [inline test-engineering/contract-tests/README.md:60]: thank you for documenting this 🙂 now my newbie docker self can stop editing docker-compose and accidentally commit it now 🙃

**Review by tiftran — APPROVED** [review]
Looks good to me!

**Pull request merged**